File: server/model/role.go
package model

type Role int

const (
	Admin Role = iota + 1
	Author
)

func (r Role) String() string {
	switch r {
	case Admin:
		return "Admin"
	case Author:
		return "Author"
	default:
		return "Author"
	}
}

func (r Role) Value() int {
	return int(r)
}

File: client/src/models/role.ts
export enum Role {
    Admin = 1,
    Author = 2
}

export const RoleToString = (role: Role) => {
    switch (role) {
        case Role.Admin:
            return "Admin";
        case Role.Author:
            return "Author";
        default:
            return "Unknown";
    }
}

export const RoleToValue = (role: Role) => {
    switch (role) {
        case Role.Admin:
            return 1;
        case Role.Author:
            return 2;
        default:
            return 0;
    }
}